### PR TITLE
Add .NET 6 Android target

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

- Add `net6.0-android` target
- Fixed android bindings and warnings in it
- Remove separate binding projects
- SDK-style projects for mono builds
- Common properties moved to `Directory.Build.props`
- Reuse same folder to multiple projects to avoid extra links and configs
- Remove usage of unsupported shared projects

## Related PRs or issues

[AB#94288](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/94288)

### CODE REVIEW

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Apple/Analytics.cs [hunk 1]
```diff
@@ -64,10 +64,10 @@ internal static void Resume()
             MSACAnalytics.Resume();
         }
 
-        /// <summary>
-        /// Enable or disable automatic page tracking.
-        /// Set this to false to if you plan to call <see cref="TrackPage"/> manually.
-        /// </summary>
+        ///// <summary>
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Apple/Analytics.cs:67]: ?

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Shared/NamespaceDoc.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Microsoft.AppCenter.Analytics
     /// <summary>
     /// Analytics SDK module.
     /// </summary>
-    internal class NamespaceDoc
+    class NamespaceDoc
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Shared/NamespaceDoc.cs:9]: why?

File: Tests/Microsoft.AppCenter.Analytics.Test/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+[assembly: ComVisible(false)]
+[assembly: Guid("c5dac35e-b902-4e24-a8f7-963ad0f85f7a")]
```
Comment (DmitriyKirakosyan) [inline Tests/Microsoft.AppCenter.Analytics.Test/Properties/AssemblyInfo.cs:8]: Guid? Scanning tools may be not happy :)

File: Tests/Microsoft.AppCenter.Test/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+[assembly: ComVisible(false)]
+[assembly: Guid("e76dc547-2b63-4911-971c-767587ebfab6")]
+
+[assembly: InternalsVisibleTo("Microsoft.AppCenter.Test.Windows, PublicKey=002400000480000094000000060200000024000052534131000400000100010055c4e2f76a6f3430448b1fd5b9ced790181e698a86759ece168bd955efc4297c9f89a303204019a9d2e8e92d204ba87e4825b36f8ba08113dc7297dcebe3d2bc15fabeae1d8c71d69769adedbc37ba7e197efc537cac2d477772ab38c4d4ccee45ddf99ce4343e9b665b663280c4dae2520b508bc7de0faf1978934f094d68e3")]
```
Comment (DmitriyKirakosyan) [inline Tests/Microsoft.AppCenter.Test/Properties/AssemblyInfo.cs:10]: duplicated

File: nuget/MacAppCenter.nuspec [hunk 5]
```diff
@@ -15,11 +15,21 @@
...
+      </group>
+      <group targetFramework="MonoAndroid10.0">
+      </group>
+      <group targetFramework="Xamarin.iOS10">
+      </group>
+      <group targetFramework="Xamarin.Mac20">
+      </group>
+      <group targetFramework="net6.0-android29.0">
+      </group>
+      <group targetFramework="net6.0-ios9.0">
```
Comment (DmitriyKirakosyan) [inline nuget/MacAppCenter.nuspec:29]: why ios9 with net6?

File: scripts/configuration/AssemblyGroup.cake [hunk 6]
```diff
@@ -36,7 +36,7 @@ public class AssemblyGroup
         var platformString = Statics.Context.IsRunningOnUnix() ? "mac" : "windows";
         if (buildGroup != null)
         {
-            Download = (buildGroup != platformString);
+            Download = buildGroup != platformString || Statics.Context.Argument<bool>("MainPackage", false);
```
Comment (DmitriyKirakosyan) [inline scripts/configuration/AssemblyGroup.cake:39]: why so complicated?

File: version.cake [hunk 7]
```diff
@@ -328,6 +329,11 @@ void UpdateNewProjSdkVersion(string newVersion, string newFileVersion)
             UpdateNewProjVersion(file, newVersion, newFileVersion);
         }
     }
+    var propsFiles = GetFiles("**/Directory.Build.props");
+    foreach (var file in propsFiles)
+    {
+            UpdateNewProjVersion(file, newVersion, newFileVersion);
```
Comment (DmitriyKirakosyan) [inline version.cake:335]: looks like formatting is not correct.

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Apple/Analytics.cs (hunk 1 above)
Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Apple/Analytics.cs:67]: It was XML doc of unexisting method in compilers point of view. There is similar approach a few lines below

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Shared/NamespaceDoc.cs (hunk 2 above)
Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.Shared/NamespaceDoc.cs:9]: There is a few such classes. It was only one with explicit one, so it's harmonization and aligning with documentation

File: Tests/Microsoft.AppCenter.Test/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (MatkovIvan) [reply Tests/Microsoft.AppCenter.Test/Properties/AssemblyInfo.cs:10]: With what?

File: nuget/MacAppCenter.nuspec (hunk 5 above)
Comment (MatkovIvan) [reply nuget/MacAppCenter.nuspec:29]: minimal supported target version

File: scripts/configuration/AssemblyGroup.cake (hunk 6 above)
Comment (MatkovIvan) [reply scripts/configuration/AssemblyGroup.cake:39]: To avoid recompiling on packaging stage

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**